src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -18,39 +18,33 @@
 package org.apache.cassandra.db.virtual;
 
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
-import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
 import com.google.common.collect.ImmutableMap;
 
-import org.apache.cassandra.config.Config;
 import org.apache.cassandra.config.DatabaseDescriptor;
-import org.apache.cassandra.config.Loader;
-import org.apache.cassandra.config.Properties;
-import org.apache.cassandra.config.Replacement;
-import org.apache.cassandra.config.Replacements;
 import org.apache.cassandra.db.DecoratedKey;
 import org.apache.cassandra.db.marshal.UTF8Type;
 import org.apache.cassandra.dht.LocalPartitioner;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.exceptions.PropertyNotFoundException;
 import org.apache.cassandra.schema.TableMetadata;
 import org.apache.cassandra.service.ClientWarn;
-import org.yaml.snakeyaml.introspector.Property;
 
-final class SettingsTable extends AbstractVirtualTable
+import static org.apache.cassandra.config.DatabaseDescriptor.getStringProperty;
+import static org.apache.cassandra.cql3.statements.RequestValidations.invalidRequest;
+import static org.apache.cassandra.utils.FBUtilities.runExceptionally;
+
+final class SettingsTable extends AbstractMutableVirtualTable
```

Comment:
where are you handing yaml string conversions (to / from)?  I don't see anything do that; did you push that out of this patch?  I saw you did a large refactor to YamlConfigLoader but that isn't used here?

Reply:
Originally I used them in the `DatabaseDescriptor` class because JMX and SettingsTable have the same type conversions, so it made sense to keep them there. But as we discussed, we currently moving them to SettingsTable only and will take care (if any) about JMX later.